adding inbound=true to auto login redirects

ericbuckley opened the pull request:
## Description
To track the difference between a manual (user clicks the login button) and an auto (on load we detect an active SSOe session and automatically redirect) login, add the query parameter `inbound=true` to the API call.

## Testing done
- updated existing auto login unit test to show the query param is there

closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/9651

ericbuckley committed:
adding ?inbound=true to auto login redirects

patrickvinograd reviewed (approved):
LGTM. I think we've added enough complexity that we need to make sure the new contract between vets-website and vets-api is documented. I know there are various documentations of the overall SAML flow in va.gov-team, but I'm not sure if there's a good doc that describes the initial "login" requests or the final callback from vets-api -> vets-website.

bkjohnson commented on src/platform/user/authentication/utilities.js:
```diff
@@ -20,13 +20,14 @@ function sessionTypeUrl(
   version = 'v0',
   application = null,
   to = null,
+  queryParams = null,
```
Is there a reason why you have this default set to null instead of an empty object literal?

bkjohnson commented on src/platform/user/authentication/utilities.js:
```diff
@@ -20,13 +20,14 @@ function sessionTypeUrl(
       : `${environment.API_URL}/sessions`;
 
-  const searchParams = new URLSearchParams();
+  const searchParams = new URLSearchParams(queryParams || {});
```
With the default parameter set to `{}` instead of null the `|| {}` wouldn't be needed here.

ericbuckley committed:
use the empty object as the default queryParams parameter

ericbuckley committed:
Merge branch 'master' into eb-inbound-ssoe-param

ericbuckley commented:
> LGTM. I think we've added enough complexity that we need to make sure the new contract between vets-website and vets-api is documented. I know there are various documentations of the overall SAML flow in va.gov-team, but I'm not sure if there's a good doc that describes the initial "login" requests or the final callback from vets-api -> vets-website.

I agree we need to work on a document to better describe all the "workarounds" we've introduced
